#! /bin/sh -e
## 120137_ftbfs_hppa_gcc3.0.dpatch by James Troup <troup@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix FTBFS on hppa with gcc-3.0.

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0

diff -x debian -ruN old/hfsplus-1.0.4/src/glob.c hfsplus-1.0.4/src/glob.c
--- old/hfsplus-1.0.4/src/glob.c	2002-03-05 19:50:29.000000000 +0000
+++ hfsplus-1.0.4/src/glob.c	2006-09-13 20:52:15.000000000 +0000
@@ -48,7 +48,7 @@
 
 	switch (*pat)
 	{
-	    case NULL:	// pattern at end
+	    case 0:	// pattern at end
 		return (!*str); // String at end ?
 
 	    case '*':  // match all
